junto-rust/zomes/core/code/src/expressions/channel.rs

```diff
@@ -0,0 +1,123 @@
+    };
+    let public_den = app_definitions::Channel{ //Create public den - personal expression place viewable by everyone
+        parent: username_address.clone(),
+        name: (first_name.clone()  + "'s Den").to_string(),
+        privacy: app_definitions::Privacy::Public,
+        channel_type: app_definitions::ChannelType::Den
+    };
+    let private_entry = Entry::App("channel".into(), private_den.into());
+    let shared_entry = Entry::App("channel".into(), shared_den.into());
+    let public_entry = Entry::App("channel".into(), public_den.into());
+    let private_den_address: Address;
+    let shared_den_address: Address;
+    let public_den_address: Address;
+    match commit_den(&private_entry, &username_address){
+        Ok(address) => private_den_address = address,
+        Err(err) => return Err(ZomeApiError::from(err))
+    };
+
+    match commit_den(&shared_entry, &username_address){
+        Ok(address) => shared_den_address = address,
+        Err(err) => return Err(ZomeApiError::from(err))
+    };
+
+    match commit_den(&public_entry, &username_address){
+        Ok(address) => public_den_address = address,
+        Err(err) => return Err(ZomeApiError::from(err))
+    };
+
+    Ok(json!({"private_den_address": private_den_address, "shared_den_address": shared_den_address, "public_den_address": public_den_address}))
+}
+
+pub fn is_den_owner(den: Address, user: Address) -> ZomeApiResult<bool>{
+    match utils::get_links_and_load_type::<String, app_definitions::UserName>(&den, "owner".to_string()){
+        Ok(result_vec) => return Ok(result_vec[0].address == user),
+        Err(err) => return Err(ZomeApiError::from(err))
+    }
+}
+
+pub fn create_collective_channel(context: &Address) -> ZomeApiResult<Address> {
+    let channel_entry: app_definitions::Channel;
+    if context == &Address::from(hdk::api::DNA_ADDRESS.to_string()) {
+        channel_entry = app_definitions::Channel{parent: context.clone(), name: "Collective".to_string(), 
+                                        privacy: app_definitions::Privacy::Public, channel_type: app_definitions::ChannelType::Tag};
+    } else { 
+        let user_name_address = user::get_user_username_address_by_agent_address()?;
+        if (group::is_group_member(context.clone(), user_name_address.clone())? == true) | (group::is_group_owner(context.clone(), user_name_address.clone())? == true) {
+            channel_entry = app_definitions::Channel{parent: context.clone(), name: "Collective".to_string(), 
+                                        privacy: app_definitions::Privacy::Shared, channel_type: app_definitions::ChannelType::Tag};
+        } else {
+            return Err(ZomeApiError::from("You are not a member/owner of given channel".to_string()))
+        };
+    };
+    let channel_collective_entry = Entry::App("channel".into(), channel_entry.into());
+    let collective_address = hdk::entry_address(&channel_collective_entry)?;
```

Comment:
It is probably safe to just add the `channel_collective_entry` every time without worrying if it exists or not. If it does already exist it won't be added again so this saves you a network lookup